opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -16,82 +16,138 @@
 from collections import OrderedDict
 from logging import getLogger
 from math import inf
+from threading import Lock
+from typing import Generic, Optional, Sequence, TypeVar
 
-from opentelemetry._metrics.instrument import _Monotonic
+from opentelemetry.sdk._metrics.measurement import Measurement
+from opentelemetry.sdk._metrics.point import Gauge, Histogram, PointT, Sum
 from opentelemetry.util._time import _time_ns
 
+# FIXME this is being copied directly from
+# opentelemetry.proto.metrics.v1.metrics_pb2. The only reason for doing so is
+# to avoid havinv protobuf as a indirect dependency in the SDK. This
+# duplication of code is not ideal.
+
+AGGREGATION_TEMPORALITY_UNSPECIFIED = 0
+AGGREGATION_TEMPORALITY_DELTA = 1
+AGGREGATION_TEMPORALITY_CUMULATIVE = 2
+
+_PointVarT = TypeVar("_PointVarT", bound=PointT)
+
 _logger = getLogger(__name__)
 
 
-class Aggregation(ABC):
+class Aggregation(ABC, Generic[_PointVarT]):
+    def __init__(self, is_monotonic: bool):
+        self._value = None
+        self._is_monotonic = is_monotonic
+        self._lock = Lock()
+
+    @property
+    def is_monotonic(self):
+        return self._is_monotonic
+
     @property
     def value(self):
-        return self._value  # pylint: disable=no-member
+        return self._value
 
     @abstractmethod
-    def aggregate(self, value):
+    def aggregate(self, measurement: Measurement) -> None:
         pass
 
     @abstractmethod
-    def make_point_and_reset(self):
-        """
-        Atomically return a point for the current value of the metric and reset the internal state.
-        """
-
+    def _collect(self) -> Optional[_PointVarT]:
+        pass
 
-class SumAggregation(Aggregation):
-    """
-    This aggregation collects data for the SDK sum metric point.
-    """
 
-    def __init__(self, instrument):
+class SynchronousSumAggregation(Aggregation[Sum]):
+    def __init__(self, is_monotonic: bool):
+        super().__init__(is_monotonic)
         self._value = 0
+        self._start_time_unix_nano = _time_ns()
```

Comment:
We should actually share the same start time for all cumulative metrics in the sdk i think. Can we pass in the start time as a constructor param?

Reply:
We can, yes, but that would mean passing the same value over and over again, right? When will `start_time_unix_nano` be set? When the SDK is "loaded"? When we make the first collection? :shrug: 

Whatever that moment in time may be, we can just save it as an attribute of the `Aggregation` class so that every aggregation has access to it.

Reply:
It will be the start time of the current collection since the Aggregations should report delta points. So actually could just pass the current start/end interval to `collect()` now that I think about it

Reply:
sorry, the start/end interval? don't we just need the start time?

Reply:
This is not really required but it's nicest for backends if all the `start_time_unix_nano` and `time_unix_nano` for a given reader's collection are the same across all the time series. For example, if you want to group by a label in your backend, having time skew can make things look funny.

Reply:
Hm, can we add this in a later PR? Kinda sounds like we need to add some reader implementation to test this.

Reply:
sure

Reply:
@aabmass @ocelotl 
I'm not too sure what start/end time represents. 

 >It will be the start time of the current collection since the Aggregations should report delta points.

Does that mean start time is recorded when `MeasurementConsumer` calls `collect`, which is then passed onto `MetricReader` -> `View` -> `Aggregation`? How will the end time be set?

Anyways, I've created this [issue](https://github.com/open-telemetry/opentelemetry-python/issues/2352) to discuss, don't want to be blocking this PR. @ocelotl  feel free to remove anything `time` related from this PR.